frontend/src/components/pages/MainPage.tsx

```diff
@@ -7,36 +7,65 @@ import { EventsPage } from './EventsPage';
 import { SlackPage } from './SlackPage';
 import { VideoPage } from './VideoPage';
 
+interface PageAbstract {
+  component: ReactElement;
+  duration: number;
+}
+
+interface PageSpecification extends PageAbstract {
+  priority: () => number;
+}
+
+interface Page extends PageAbstract {
+  probability: number;
+}
+
+function preparePageSpecifications(pages: PageSpecification[]): Page[] {
+  const totalPriority = pages.map(page => page.priority()).reduce((a, b) => a + b, 0)
+
+  return pages.map((page) => {
+    const mappedPage = page as any
```

Comment:
Hvorfor `any` her?

Reply:
Fordi TypeScript ikke vil la meg slette en property som ikke er Optinal for typen. Så da gjør jeg den om til any før jeg endrer på selve objektet.
Er nok ikke den beste måten å gjøre det på, men ville ikke manuelt gjenta alle propertiene i det nye objektet i tilfelle strukturen til PageSpecification endres igjen.
Om du har en bedre måte å gjøre det på, si ifra!

Reply:
```
return pages.map((page) => {
    const { priority, ...rest } = page;
    return {
      ...rest,
      probability: priority() / totalPriority
    } as Page;
  });
```